var/spack/repos/builtin/packages/py-celery/package.py

```diff
@@ -65,10 +67,29 @@ class PyCelery(PythonPackage):
     depends_on("py-click-repl@0.2.0:", when="@5.2.0:", type=("build", "run"))
     depends_on("py-pytz@2019.3:", type=("build", "run"))
     depends_on("py-pytz@2021.3:", type=("build", "run"), when="@5.2.3")
-    depends_on("py-billiard@3.6.3.0:3", type=("build", "run"))
+    depends_on("py-billiard@3.6.3.0:3", type=("build", "run"), when="@:5.0.0")
     depends_on("py-billiard@3.6.4.0:3", type=("build", "run"), when="@5.2.3")
     depends_on("py-kombu@4.6.11", when="@4.3.0:4", type=("build", "run"))
     depends_on("py-kombu@5.0.0:", when="@5.0.0:5.0", type=("build", "run"))
     depends_on("py-kombu@5.2.3:5", when="@5.2.0:5.2", type=("build", "run"))
     depends_on("py-vine@1.3.0", when="@4.3.0:4", type=("build", "run"))
     depends_on("py-vine@5.0.0:5", when="@5.0.0:5", type=("build", "run"))
+
+    with when("@5.2.3:5"):
+        depends_on("py-cryptography", type=("build", "run"))
+
+    with when("@5.3.4:5"):
+        depends_on("py-billiard@4.1:4", type=("build", "run"))
+        depends_on("py-kombu@5.3.2:5", type=("build", "run"))
+        depends_on("py-vine@5.0.0:5", type=("build", "run"))
+        depends_on("py-click@8.1.2:8", type=("build", "run"))
+        depends_on("py-click-didyoumean@0.3.0:", type=("build", "run"))
+        depends_on("py-click-repl@0.2.0:", type=("build", "run"))
+        depends_on("py-click-plugins@1.1.1:", type=("build", "run"))
+        depends_on("py-backports-zoneinfo@0.2.1:", when="^python@:3.9", type=("build", "run"))
+        depends_on("py-pytz@2022.7:", type=("build", "run"))
+        depends_on("py-python-dateutil@2.8.2:", type=("build", "run"))
+        # https://github.com/celery/celery/blob/v5.3.4/requirements/extras/sqlalchemy.txt
+        depends_on("py-sqlalchemy@1.4.48:2.0", when="+sqlalchemy", type=("build", "run"))
+        # https://github.com/celery/celery/blob/v5.3.4/requirements/extras/auth.txt
+        depends_on("py-cryptography@41.0.3:", type=("build", "run"))
```

Comment:
```suggestion
        depends_on("py-cryptography@41.0.3", when="+redis", type=("build", "run"))
```

Reply:
Why this change?

Reply:
Ah sorry, should be:
```suggestion
        depends_on("py-cryptography@41.0.3", when="+auth", type=("build", "run"))
```

Reply:
Or just remove it since it's not required